src/phoenix/session/client.py

```diff
@@ -249,6 +264,142 @@ def log_traces(self, trace_dataset: TraceDataset, project_name: Optional[str] =
                 },
             ).raise_for_status()
 
+    def upload_dataset_csv(
+        self,
+        file: Union[str, Path],
```

Comment:
It would change the ergonomics a bit, but could we take in a file handle here rather than a file path? E.g.,

```
with open("/path/to/csv") as f:
    client.upload_dataset_csv(f, input_keys=..., ...)
```

This would be more flexible since people could use buffers in addition to files on disk, etc.